src/VolleyManagement.Backend/VolleyManagement.Services/GameService.cs

```diff
@@ -499,62 +499,81 @@ select game
-                           string.Format(
-                           Resources.SameGameInRound,
-                           game.HomeTeamName,
-                           game.AwayTeamName,
-                           game.Round.ToString()));
-                    }
+                    ValidateAreSameTeamsInGames(game, newGame, tournamentScheduleInfo);
                 }
                 else if (GameValidation.IsTheSameTeamInTwoGames(game, newGame))
                 {
-                    if (GameValidation.IsFreeDayGame(newGame))
-                    {
-                        if (tournamentScheduleInfo.Scheme != TournamentSchemeEnum.PlayOff
-                            && game.HomeTeamId != newGame.HomeTeamId
-                            && game.AwayTeamId != newGame.HomeTeamId)
-                        {
-                            throw new ArgumentException(
-                                Resources.SameFreeDayGameInRound);
-                        }
-                    }
-                    else
-                    {
-                        string oppositeTeam;
+                    ValidateIsTheSameTeamInTwoGames(game, newGame, tournamentScheduleInfo);
+                }
+            }
+        }
 
-                        if (game.HomeTeamId == newGame.HomeTeamId
-                            || game.HomeTeamId == newGame.AwayTeamId)
-                        {
-                            oppositeTeam = game.HomeTeamName;
-                        }
-                        else
-                        {
-                            oppositeTeam = game.AwayTeamName;
-                        }
+        private static void ValidateAreSameTeamsInGames(
+            GameResultDto game,
+            Game newGame,
+            TournamentScheduleDto tournamentScheduleInfo)
+        {
+            string errorMessage = null;
+            if (GameValidation.IsFreeDayGame(newGame))
+            {
+                if (tournamentScheduleInfo.Scheme != TournamentSchemeEnum.PlayOff)
+                {
+                    errorMessage = Resources.SameFreeDayGameInRound;
+                }
+                else
+                {
+                    errorMessage = string.Format(
+                        Resources.SameTeamInRound,
+                        game.HomeTeamId);
+                }
+            }
+            else
+            {
+                errorMessage = String.Format(
+                    Resources.SameGameInRound,
+                    game.HomeTeamName,
+                    game.AwayTeamName,
+                    game.Round.ToString());
+            }
+            throw new ArgumentException(errorMessage);
+        }
 
-                        throw new ArgumentException(
-                          string.Format(
-                          Resources.SameTeamInRound,
-                                 oppositeTeam));
-                    }
+        private static void ValidateIsTheSameTeamInTwoGames(
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') The Cyclomatic Complexity of this method is 11 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1541)